integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/multitenancy/CustomMongoDatabaseResolver.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.it.mongodb.panache.multitenancy;
+
+import javax.enterprise.context.RequestScoped;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.mongodb.panache.common.MongoDatabaseResolver;
+import io.vertx.ext.web.RoutingContext;
+
+@Unremovable
```

Comment:
Should not be needed, you can make all beans unremovable from the deployment module.

Reply:
May be it sounds a dumb question, Are you saying the deployment module from mongo-panache or integration test?